src/cargo/core/workspace.rs

```diff
@@ -466,19 +458,65 @@ impl<'cfg> Workspace<'cfg> {
     }
 
     /// Returns an iterator over all packages in this workspace
-    pub fn members<'a>(&'a self) -> Members<'a, 'cfg> {
-        Members {
-            ws: self,
-            iter: self.members.iter(),
-        }
+    pub fn members(&self) -> impl Iterator<Item = &Package> {
+        let packages = &self.packages;
+        self.members
+            .iter()
+            .filter_map(move |path| match packages.get(path) {
+                &MaybePackage::Package(ref p) => Some(p),
+                _ => None,
+            })
+    }
+
+    /// Returns a mutable iterator over all packages in this workspace
+    pub fn members_mut(&mut self) -> impl Iterator<Item = &mut Package> {
+        let packages = &mut self.packages.packages;
+        let members: HashSet<_> = self
+            .members
+            .iter()
+            .map(|path| path.parent().unwrap().to_owned())
+            .collect();
+
+        packages.iter_mut().filter_map(move |(path, package)| {
+            if members.contains(path) {
```

Comment:
Could this use `package.get_mut` like above uses `packages.get`?

(same for `default_members_mut` below)

Reply:
no, the `get_mut` lifetime gets (rightfully) in the way.